categorize_adjectives/bert_similarity_calc.py

```diff
@@ -0,0 +1,60 @@
+"""Generate dictionary of distances between each pair of adjective"""
+import os
+import pickle
+from libraries.bert_embeddings import BertEmbedding
+from scipy.spatial.distance import cosine
+
+def get_most_frequent_adjs(noun_to_adj, num_adjs):
+    """Generate the list of most frequent num_adjs in noun_to_adj dictionary"""
+    selected_adj = []
+    adj_count = {}
+    for noun in noun_to_adj.keys():
+        for adj, _ in noun_to_adj[noun]:
+            try:
+                adj_count[adj] += -1
+            except KeyError:
+                adj_count[adj] = -1
+    sorted_adjs = sorted(adj_count.items(), key=lambda kv: (kv[1], kv[0]))[:num_adjs]
+    for adj, _ in sorted_adjs:
+        selected_adj.append(adj)
+    return selected_adj
+
+def generate_distance_metric(adj_list):
+    """generate the distance metric between each pair of adjective from adj_list
+
+    First it calls the get_embeddings library to get embeddings of each adjective
+    and then takes cosine distance between each pair of adjective from adj_list
+
+    returns a dicitonary that contains tuple of two words and point to their distance
+    """
+    calculated_dis = {}
+    bert_embeddings_gen = BertEmbedding()
+    tuple_adj = []
+    for adj in adj_list:
+        tuple_adj.append((adj,))
+    embeddings = bert_embeddings_gen.get_embeddings(tuple_adj, layer=-2)
+    for i, adj_1 in enumerate(adj_list):
+        for j, adj_2 in enumerate(adj_list):
+            if j < i:
```

Comment:
You can add enumerate(adj_list, i+1) in the above line instead.
https://docs.python.org/3/library/functions.html#enumerate